### PR TITLE
[main][feat] biswap deployment script

### PR DESCRIPTION
**Pull request by sora-alpaca**
## Description
Deployment script for Biswap strategies and worker

## Why?
To deploy contracts, we have to prepare deployment scripts for deployer.

## ChangeLogs:
- prepare biswap share strategies deployment script
- prepare biswap vault strategies deployment script
- prepare biswap worker deployment script
- modify Config interface to support Biswap config
- modify mainnet.json interface
- modify testnet.json interface

### Link to story (if available)
https://100x.atlassian.net/browse/ALPACA-3963

### CODE REVIEW

File: .mainnet.json [hunk 1]
```diff
@@ -1106,6 +1119,13 @@
       "StrategyWithdrawMinimizeTrading": "0x94f5E884e11fC054Ca0b332e3a8773750BDBe242",
       "StrategyPartialCloseLiquidate": "0xa6b2a3Fdd415Ca304a36909a35f974B670960843",
       "StrategyPartialCloseMinimizeTrading": "0xDD8A4302299115226676F943E01306107DD98495"
+    },
+    "Biswap": {
+      "StrategyAddBaseTokenOnly": "0xB50258Ce105DCB0476f4366095729788FbAE8fA7",
```
Comment (spicysquid168) [inline .mainnet.json:1124]: Put address(0) first

File: .mainnet.json [hunk 2]
```diff
@@ -1140,7 +1160,8 @@
         "Pancakeswap": "0xB9B8766B65636779C3B169B9a18e0A708F91c610",
         "Waultswap": "0xA7559bB0235a1c6003D0E48d2cFa89a6C8748439",
         "PancakeswapSingleAsset": "0xB0951EB5eCd9948AAA8Eb76D1061361F592BA029",
-        "Mdex": "0xD8A128e5712D9b240811a0D55D3F9CA8E6599B7A"
+        "Mdex": "0xD8A128e5712D9b240811a0D55D3F9CA8E6599B7A",
+        "Biswap": "0xf5295390f831b6E3019955154e420E231Cf9a56d"
```
Comment (spicysquid168) [inline .mainnet.json:1164]: Put address(0) as a placeholder

File: .mainnet.json [hunk 3]
```diff
@@ -1907,6 +1928,23 @@
             "StrategyPartialCloseLiquidate": "0x4362635a0b2F8dF98cDE4Cdde5F5f3bE96f45316",
             "StrategyPartialCloseMinimizeTrading": "0x8dcEC5e136B6321a50F8567588c2f25738D286C2"
           }
+        },
+        {
+          "name": "USDT-WBNB BiswapWorker",
+          "address": "0xaB3A0282f9d0C10d29e0b8c1336b24c0AD8678dF",
```
Comment (spicysquid168) [inline .mainnet.json:1934]: Put address(0) as a placeholder

File: .mainnet.json [hunk 4]
```diff
@@ -3084,6 +3126,23 @@
             "StrategyPartialCloseLiquidate": "0x4362635a0b2F8dF98cDE4Cdde5F5f3bE96f45316",
             "StrategyPartialCloseMinimizeTrading": "0x8dcEC5e136B6321a50F8567588c2f25738D286C2"
           }
+        },
+        {
+          "name": "WBNB-USDT BiswapWorker",
+          "address": "0xb7Aa8DC3EDCA9D9275871b159B35819A43aa3548",
```
Comment (spicysquid168) [inline .mainnet.json:3132]: Put address(0) as a placeholder

File: deploy/exec/workers/deploy/biswap-workers03.ts [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+      okStrats.push(workerInfos[i].PARTIAL_CLOSE_LIQ_STRAT_ADDR);
+    }
+    if (validateAddress(workerInfos[i].PARTIAL_CLOSE_MINIMIZE_STRAT_ADDR)) {
+      okStrats.push(workerInfos[i].PARTIAL_CLOSE_MINIMIZE_STRAT_ADDR);
+    }
+    await biswapWorker03.setStrategyOk(okStrats, true, { nonce: nonce++ });
+    console.log("✅ Done");
+
+    console.log(`>> Whitelisting a worker on strats`);
+    const addStrat = BiswapStrategyAddBaseTokenOnly__factory.connect(workerInfos[i].ADD_STRAT_ADDR, deployer);
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/biswap-workers03.ts:218]: I think here can make it cleaner by loop through `okStrats` and `setWorkersOk` for each `okStrats`

File: utils/address.ts [hunk 6]
```diff
@@ -1,3 +1,9 @@
+import { zeroAddress } from "ethereumjs-util";
```
Comment (spicysquid168) [inline utils/address.ts:1]: use `ethers` instead => `ethers.constant.AddressZero`

File: deploy/exec/workers/deploy/biswap-workers03.ts [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+      console.log("✅ Done");
+    } else {
+      console.log(">> Linking VaultConfig with WorkerConfig");
+      (
+        await vaultConfig.setWorkers([biswapWorker03.address], [workerInfos[i].WORKER_CONFIG_ADDR], {
+          nonce: nonce++,
+        })
+      ).wait(3);
+      console.log("✅ Done");
+    }
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/biswap-workers03.ts:317]: I think you can add a newly deploy worker to `config` right here and then write it into `.mainnet.json` file

**Review by spicysquid168 — DISMISSED** [review]

**Comment by sora-alpaca** [comment]
NOTE: waiting for deployment on mainnet before merge

**Review by abc-alpaca — APPROVED** [review]

File: deploy/exec/workers/deploy/biswap-workers03.ts [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+        StrategyAddAllBaseToken: workerInfos[i].ADD_STRAT_ADDR,
+        StrategyLiquidate: workerInfos[i].LIQ_STRAT_ADDR,
+        StrategyAddTwoSidesOptimal: workerInfos[i].TWO_SIDES_STRAT_ADDR,
+        StrategyWithdrawMinimizeTrading: workerInfos[i].MINIMIZE_TRADE_STRAT_ADDR,
+        StrategyPartialCloseLiquidate: workerInfos[i].PARTIAL_CLOSE_LIQ_STRAT_ADDR,
+        StrategyPartialCloseMinimizeTrading: workerInfos[i].PARTIAL_CLOSE_MINIMIZE_STRAT_ADDR,
+      },
+    };
+
+    config = configFileHelper.addOrSetVaultWorker(workerInfos[i].VAULT_ADDR, biswapWorkersEntity);
```
Comment (spicysquid168) [inline deploy/exec/workers/deploy/biswap-workers03.ts:221]: Move this to after all configuration done

**Review by spicysquid168 — APPROVED** [review]

**Review by jr-alpaca — APPROVED** [review]

**Pull request merged**